native/effektio/src/platform/desktop.rs

```diff
@@ -18,17 +23,102 @@ pub async fn new_client_config(base_path: String, home: String) -> Result<Client
     Ok(builder)
 }
 
-pub fn init_logging(filter: Option<String>) -> Result<()> {
+static mut FILE_LOGGER: Option<Arc<fern::ImplDispatch>> = None;
+
+// this excludes macos, because macos and ios is very much alike in logging
+
+pub fn init_logging(app_name: String, log_dir: String, filter: Option<String>) -> Result<()> {
     std::env::set_var("RUST_BACKTRACE", "1");
     log_panics::init();
 
-    let logger = if let Some(log_str) = filter {
-        Logger::try_with_env_or_str(log_str)?
-    } else {
-        Logger::try_with_env()?
+    let log_level = match filter {
+        Some(filter) => FilterBuilder::new().parse(&filter).build(),
+        None => FilterBuilder::new().build(),
     };
-    logger
-        .adaptive_format_for_stderr(AdaptiveFormat::Detailed)
-        .start()?;
+
+    let mut path = PathBuf::from(log_dir.as_str());
+    path.push("app_");
+
+    let (level, dispatch) = fern::Dispatch::new()
+        .format(|out, message, record| {
+            out.finish(format_args!(
+                "{}[{}][{}] {}",
+                chrono::Local::now().format("[%Y-%m-%d][%H:%M:%S]"),
+                record.target(),
+                record.level(),
+                message
+            ))
+        })
+        // Only log messages Info and above
+        .level(log_level.filter())
+        // Output to console
+        .chain(std::io::stdout())
+        // Output to file
+        .chain(fern::Manual::new(path, "%Y-%m-%d_%H-%M-%S%.f.log"))
+        .into_dispatch_with_arc();
+
+    if level == log::LevelFilter::Off {
+        log::set_boxed_logger(Box::new(native::NopLogger)).unwrap();
+    } else {
+        log::set_boxed_logger(Box::new(dispatch.clone())).unwrap();
+    }
+    log::set_max_level(level);
+
+    unsafe {
+        FILE_LOGGER = Some(dispatch);
+    }
+
     Ok(())
 }
+
+pub async fn report_bug(
```

Comment:
this is the same as for `android` and below, right? Please put it into a common place then as one function and import it.  Having three copies will lead to the code diverging...

Reply:
Fixed by https://github.com/acterglobal/a3/pull/488/commits/3ee79ed77286ca599993d15ba9df4e092e7f77f6